src/App.js

```diff
@@ -1,16 +1,35 @@
-import React from 'react';
+import React, {useState} from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
 
+
+
+
+
+  
 const App = () => {
-  return (
+  const [chatData, setChatData] = useState(chatMessages);
+  const handleToggleLiked = (updatedEntry) => {
+    const newChatData = chatData.map((entry) => {
+        if (entry.id === updatedEntry.id) {
+          return updatedEntry;
+        } else {
+          return entry
+        }
+  });
+setChatData(newChatData)}
```

Comment:
handleToggleLiked executes when the like button on a ChatEntry component is clicked. We really only need to update the heart on a ChatEntry, not the entire message. How would you refactor handleToggleLiked so that you aren't updating the entire ChatEntry object?

You can check out this [implementation](https://github.com/ada-c17/react-chatlog/blob/c17/solution/src/App.js#L22-L30) for an idea of how this refactor would look